### PR TITLE
[s] Improved chem grenade logging

### LINKED ISSUE #38407: nade/smoke logging can be abused to get somebody else banned.
The prime message for chem nades does not say what chemicals it has, the explosion/smoke message uses last key which can be incorrect and the smoke message says `associated key` not `last touched by` so admins think its a fool proof message of who space lubed an entire area when it is not.

These combined were being used to get other people banned for space lubing the ball's dance floor.

A pr is test merged that disables (`del()`s) all grenades because of the severity of this bug.

### PR DESCRIPTION
**Pull request by ShizCalev**
Fixes #38407

### CODE REVIEW

File: code/game/objects/items/grenades/chem_grenade.dm [hunk 1]
```diff
@@ -75,6 +75,10 @@
 					return
 				to_chat(user, "<span class='notice'>You add [I] to the [initial(name)] assembly.</span>")
 				beakers += I
+				for(var/datum/reagent/R in I.reagents.reagent_list)
+					var/reagent_vol = R.volume
+					add_logs(user, src, "inserted [I]" additional = "[reagent_vol] units of [R] inside.")
```
Comment (Jordie0608) [inline code/game/objects/items/grenades/chem_grenade.dm]: Missing a comma here before `additional`

**Review by Jordie0608 — DISMISSED** [review]

File: code/game/objects/items/grenades/chem_grenade.dm [hunk 2]
```diff
@@ -111,6 +115,12 @@
 		if(beakers.len)
 			for(var/obj/O in beakers)
 				O.forceMove(drop_location())
+				if(!O.reagents)
+					continue
+				for(var/datum/reagent/R in O.reagents.reagent_list)
+					var/reagent_vol = R.volume
+					add_logs(user, src, "removed [O]" additional = "[reagent_vol] units of [R] inside.")
```
Comment (Jordie0608) [inline code/game/objects/items/grenades/chem_grenade.dm]: Here too.

**Review by Cyberboss — DISMISSED** [review]

**Comment by Iamgoofball** [comment]
I could of sworn I added this ages ago

File: code/game/objects/items/grenades/chem_grenade.dm [hunk 3]
```diff
@@ -155,6 +165,19 @@
...
+	for(var/obj/exploded_beaker in beakers)
+		if(!exploded_beaker.reagents)
+			continue
+		for(var/datum/reagent/R in exploded_beaker.reagents.reagent_list)
+			var/reagent_volume = R.volume
+			var/message = "[src] primed by [user] at [ADMIN_VERBOSEJMP(T)] contained [reagent_volume] units of [R]."
+			GLOB.bombers += message
+			message_admins(message)
+			log_game("[src] primed by [user] at [AREACOORD(T)] contained [reagent_volume] units of [R].")
+			add_logs(user, src, "primed", additional = "[reagent_volume] units of [R] inside.")
```
Comment (AnturK) [inline code/game/objects/items/grenades/chem_grenade.dm]: Same as above.

File: code/game/objects/items/grenades/chem_grenade.dm [hunk 4]
```diff
@@ -111,6 +115,12 @@
 		if(beakers.len)
 			for(var/obj/O in beakers)
 				O.forceMove(drop_location())
+				if(!O.reagents)
+					continue
+				for(var/datum/reagent/R in O.reagents.reagent_list)
+					var/reagent_vol = R.volume
+					add_logs(user, src, "removed [O]", additional = "[reagent_vol] units of [R] inside.")
```
Comment (AnturK) [inline code/game/objects/items/grenades/chem_grenade.dm]: Additional -> Addition

**Review by AnturK — DISMISSED** [review]
Also don't we have some reagent list stringify helper ?

**Comment by AnturK** [comment]
After taking a glance at the code you can use `pretty_string_from_reagent_list` to squash these into single entry. Or is there some reason it's split like this.

**Comment by MrStonedOne** [comment]
@tgstation/commit-access

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**